plugins/org.python.pydev.parser/src/org/python/pydev/parser/prettyprinterv2/PrettyPrinterVisitorV2.java

```diff
@@ -122,18 +122,27 @@ public Object visitAssign(Assign node) throws Exception {
         java.util.List<ILinePart> recordChanges = null;
         org.python.pydev.shared_core.structure.Tuple<ILinePart, ILinePart> lowerAndHigher = null;
 
-        for (int i = 0; i < node.targets.length; i++) {
-            exprType target = node.targets[i];
-            if (i >= 1) { //more than one assign
-                doc.add(lowerAndHigher.o2.getLine(), lowerAndHigher.o2.getBeginCol(),
-                        this.prefs.getAssignPunctuation(), node);
+        String punctuation = this.prefs.getAssignPunctuation();
+        if (node.targets.length > 0) {
+            for (int i = 0; i < node.targets.length; i++) {
+                exprType target = node.targets[i];
+                if (i >= 1) { //more than one assign
+                    doc.add(lowerAndHigher.o2.getLine(), lowerAndHigher.o2.getBeginCol(),
+                            this.prefs.getAssignPunctuation(), node);
+                }
+                id = this.doc.pushRecordChanges();
+                target.accept(this);
+                recordChanges = this.doc.popRecordChanges(id);
+                lowerAndHigher = doc.getLowerAndHigerFound(recordChanges);
             }
+        } else {
             id = this.doc.pushRecordChanges();
-            target.accept(this);
+            node.type.accept(this);
             recordChanges = this.doc.popRecordChanges(id);
             lowerAndHigher = doc.getLowerAndHigerFound(recordChanges);
+            punctuation = this.prefs.getAssignPunctuationColon();
```

Comment:
Why get the punctuation again if it's already assigned (in the first line changed in this block)?

Reply:
The one assigned in the block first line is just getAssignPunctuation (the assign standard punctuation ' = ')

Reply:
Humm, then assign to a different variable (`puctuationColon`) to make it clear that this is a different thing.

Reply:
Actually, rereading it does make sense as it is... just rename the function as suggested above.